### PR TITLE
Faster PolygonTriangulator implemented

### PR DESCRIPTION
**Pull request by Iluvatar82**
A faster polygon triangulation algorithm was implemented.
A Demo Project was added to make use of the new Triangulator.
#322

### CODE REVIEW

**Comment by Iluvatar82** [comment]
Can anyone please give me a hint, why the check is not passing? I mean the SweepLinePolygonTriangulator is defenitely **DOES** exist in the current context, see: [AppVeyor Build Details](https://ci.appveyor.com/project/objorke/helix-toolkit/build/1.0.0-PullRequest.328+826.build.740/messages)

**Comment by JeremyAnsel** [comment]
`Geometry/SweepLinePolygonTriangulator.cs` is not referenced in `HelixToolkit.Wpf_NET40.csproj`.

**Comment by Iluvatar82** [comment]
Thank you, well I didn't check the _NET40 solutions, thank you for your help!

File: Source/Examples/WPF.SharpDX/LineShadingDemo/MainViewModel.cs [hunk 1]
```diff
@@ -52,7 +52,7 @@ public MainViewModel()
             this.SubTitle = null;
 
             // camera setup
-            this.Camera = new PerspectiveCamera { Position = new Point3D(0, 5, 5), LookDirection = new Vector3D(-0, -5, -5), UpDirection = new Vector3D(0, 1, 0) };
+            this.Camera = new PerspectiveCamera { Position = new Point3D(0, -20, 20), LookDirection = new Vector3D(-0, 5, -5), UpDirection = new Vector3D(0, 1, 0) };
```
Comment (objorke) [inline Source/Examples/WPF.SharpDX/LineShadingDemo/MainViewModel.cs]: improvement? not related to polygon triangulation..

File: Source/HelixToolkit.Wpf/Helpers/MeshBuilder.cs [hunk 2]
```diff
@@ -1966,10 +1966,10 @@ public void AddPolygon(IList<int> vertexIndices)
         }
 
         /// <summary>
-        /// Adds a polygon defined by vertex indices (uses the cutting ears algorithm).
+        /// Adds a polygon defined by vertex indices (uses the sweep line algorithm).
         /// </summary>
         /// <param name="vertexIndices">The vertex indices.</param>
-        public void AddPolygonByCuttingEars(IList<int> vertexIndices)
+        public void AddPolygonBySweepLine(IList<int> vertexIndices)
```
Comment (objorke) [inline Source/HelixToolkit.Wpf/Helpers/MeshBuilder.cs]: this is a breaking change, it would be better to create an overload

Comment (objorke) [inline Source/HelixToolkit.Wpf/Helpers/MeshBuilder.cs]: I think the method name should be `AddPolygon` and the `AddPolygonByCuttingEars` should be marked obsolete

**Comment by objorke** [comment]
very good work!
can you also update the change log?

**Comment by Iluvatar82** [comment]
Thank you! This are the changes I made now:

AddPolygonByCuttingEars is available again and marked Obsolete in the **MeshBuilder**
Changelog updated
Could not rename **AddPolygonBySweepLine** to **AddPolygon** because
this signature already exists in the MeshBuilder. Another neutral name
was found with **AddPolygonByTriangulation** (I hope that's ok)

**Comment by objorke** [comment]
thanks for the update! And thanks for contributing this to the project!

**Comment by objorke** [comment]
I´ll squash this PR

**Pull request merged**